p2p/simulations/http.go

```diff
@@ -380,6 +398,32 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
 		write("error", err.Error())
 	}
 
+	// check if filtering has been requested
+	if strfilter := req.URL.Query().Get("filter"); strfilter != "" {
+		arrfilter := strings.Split(strfilter, "-")
+		for _, f := range arrfilter {
+			fparts := strings.Split(f, ":")
+			if len(fparts) != 2 || len(fparts[0]) == 0 || len(fparts[1]) == 0 {
+				http.Error(w, "Invalid msg filter format provided", http.StatusBadRequest)
+				return
+			}
+			proto := fparts[0]
+			strcodes := strings.Split(fparts[1], ",")
+			for _, c := range strcodes {
+				if fparts[1] == "*" || fparts[1] == "-1" {
+					wildcards[fparts[0]] = struct{}{}
+				} else {
+					code, err := strconv.ParseUint(c, 10, 64)
+					if err != nil {
+						http.Error(w, "Invalid msg code for filtering provided", http.StatusBadRequest)
+						return
+					}
+					filters[MsgFilter{Proto: proto, Code: code}] = struct{}{}
+				}
+			}
+		}
+	}
```

Comment:
This filtering logic should be moved into a separate method and be documented (it isn't clear what the format of the filter string should be).